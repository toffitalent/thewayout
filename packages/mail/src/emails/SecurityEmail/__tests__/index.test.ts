import { renderTemplate } from '@test';
import { SecurityEmail } from '..';

describe('SecurityEmail', () => {
  const context = {
    user: {
      firstName: 'Test',
    },
  };

  const email = new SecurityEmail(context);

  test('renders correctly', async () => {
    expect(await email.render()).toMatchSnapshot();
    expect(renderTemplate('SecurityEmail', 'html', context)).toMatchSnapshot();
    expect(renderTemplate('SecurityEmail', 'subject', context)).toMatchSnapshot();
    expect(renderTemplate('SecurityEmail', 'text', context)).toMatchSnapshot();
  });
});
