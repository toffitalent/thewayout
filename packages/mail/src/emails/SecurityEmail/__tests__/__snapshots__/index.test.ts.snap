// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityEmail renders correctly 1`] = `
{
  "html": "pug-test-stub",
  "subject": "pug-test-stub",
  "text": "pug-test-stub",
}
`;

exports[`SecurityEmail renders correctly 2`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><meta name="viewport" content="width=device-width, initial-scale=1.0"/><meta name="x-apple-disable-message-reformatting"/><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/><meta name="color-scheme" content="light dark"/><meta name="supported-color-schemes" content="light dark"/><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Nunito:400,700&amp;display=swap"/><link rel="stylesheet" href="@app/styles/index.scss" data-inline="true"/><!--[if mso]><style type="text/css">.f-fallback {
  font-family: Arial, sans-serif;
}</style><![endif]--></head><body><span class="preheader">As requested, we've updated your email address. If you made this change, you don't need to do anything else.</span><table class="email__wrapper" width="100%" cellpadding="0" cellspacing="0" role="presentation"><tr><td align="center"><table class="email__content" width="100%" cellpadding="0" cellspacing="0" role="presentation"><tr><td class="email__masthead"><a class="f-fallback email__masthead__logo" href="http://__WEB__"><img src="@app/assets/logo.png" alt="The Way Out"/></a></td></tr><tr><td class="email__body" width="570" cellpadding="0" cellspacing="0"><table class="email__body__inner" align="center" width="570" cellpadding="0" cellspacing="0" role="presentation"><tr><td class="content"><div class="f-fallback"><h1>Hi Test</h1><p>The email address associated with your The Way Out account has been updated. If you made this change, you don't need to do anything else.</p><p>If you did not recently update your email address and believe an unauthorized person has accessed your account, please
<a href="mailto:support@thewayouthelps.com">contact support</a>.</p><p>Thanks,<br/>The The Way Out Team</p></div></td></tr></table></td></tr><tr><td><table class="email__footer" align="center" width="570" cellpadding="0" cellspacing="0" role="presentation"><tr><td class="content" align="center"><p class="f-fallback sub align-center">&copy; 2024 The Way Out. All rights reserved.</p><p class="f-fallback sub align-center">The Way Out, Inc.<br/><a href="#">PO Box 70620</a><br/><a href="#">Milwaukee, WI 53207</a></p></td></tr></table></td></tr></table></td></tr></table></body></html>"
`;

exports[`SecurityEmail renders correctly 3`] = `"Your The Way Out account email has been updated"`;

exports[`SecurityEmail renders correctly 4`] = `
"Hey Test,

The email address associated with your The Way Out account has been updated. If you made this change, you don't need to do anything else.

If you did not recently update your email address and believe an unauthorized person has accessed your account, please contact support (support@thewayouthelps.com).

Thanks,
The The Way Out Team

--
Â© 2024 The Way Out. All rights reserved.

The Way Out, Inc.
PO Box 70620
Milwaukee, WI 53207"
`;
